waspc/src/Generator/ExternalCodeGenerator/Common.hs

```diff
@@ -0,0 +1,19 @@
+module Generator.ExternalCodeGenerator.Common
+    ( ExternalCodeGeneratorStrategy(..)
+    ) where
+
+import Data.Text (Text)
+import qualified Path.Aliases as Path
+
+
+data ExternalCodeGeneratorStrategy = ExternalCodeGeneratorStrategy
```

Comment:
aha you extracted this as a strategy because of client and server differences?

Reply:
Yes! I know we will need it in both places, and it makes sense that import is not the same on both sides, and it wasn't already because of the paths, so I did it this way, better than passing bazillion parameters to the generateExternalCode function -> instead I pass this nice strategy.